apps/pg_subscriber/lib/pg_subscriptor/handler.ex

```diff
@@ -134,6 +140,31 @@ defmodule PgSubscriber.Handler do
     Logger.info(tuple_data: tuple_data)
   end
 
+  defp handle_delete!(body) do
+    Logger.info("Got DELETE msg")
+    <<_relation_oid::32, tuple_type::8, rest::binary>> = body
+    {:ok, tuple_data, <<>>} = TupleData.get_tuple_data(rest)
+
+    Logger.info(tuple_type: <<tuple_type>>)
+    Logger.info(tuple_data: tuple_data)
+  end
+
+  defp handle_truncate!(body) do
+    Logger.info("Got TRUNCATE msg")
+    <<relations_num::32, _options::8, rest::binary-size(4 * relations_num)>> = body
+    relation_oids = get_relations_oid(relations_num, rest)
+    Logger.info(relation_oids: relation_oids)
+  end
+
+  defp get_relations_oid(0, _) do
+    []
+  end
+
+  defp get_relations_oid(relations_num, msg) do
+    <<relation_oid::32, rest::binary>> = msg
+    [relation_oid | get_relations_oid(relations_num - 1, rest)]
```

Comment:
Maybe you should return `rest` as well? In case there were more fields to come?
It could then also be used for all `_relation_oid::32` parsing that way. Might come in handy if we wanted to map oid to relation or something in the future, like you wanted with `Update` messages to map to data from `Relation`  messages.

Otherwise, you do not need `relations_num` and can just match `rest == <<>>` in the base...